### DISCUSSION TITLE
cache issue about Nextra

### DISCUSSION THREAD

**Discussion by coleea** [vercel/community#2079] [Bug Report]
I deployed nextra app via vercel

This seemed to work when I first deploy. However, the bug occurred when I deploy several times. 

It shows a different menu every time when I click on the menu. I think it is a cache issue.

when I delete cache, It works fine again.

you can check this bug on [https://doc.fpis.fun/](https://doc.fpis.fun/)

click the menu on left side. then you'll know what I said

**Comment by coleea** [comment]
I found temporary solution
go to settings -> "environment variables" tab
add key name "VERCEL_FORCE_NO_BUILD_CACHE" and value "1"
save this.
then build cache will be disabled. 
re-deploy project.
problem solved.

but this is not a real solution.
Waiting for vercel team's clear solution

**Reply by DanielRudrich** [reply]
I have the same problems with several of our Nextra projects. I can confirm that setting `VERCEL_FORCE_NO_BUILD_CACHE ` to `1` solves this issue, but doesn't seem to be very elegant.

**Reply by sidpremkumar** [reply]
Same here, only `VERCEL_FORCE_NO_BUILD_CACHE` worked. The postbuild solution presented below did not work for me.

**Comment by marcklingen** [comment]
I have the same issue, `VERCEL_FORCE_NO_BUILD_CACHE=1` solved it but led to long build times as dependencies were not cached

I found removing `.next/cache` postbuild to be a great alternative solution ([commit in our docs repo](https://github.com/langfuse/langfuse-docs/pull/32/commits/a7018fd2c2818d59dee7d604bc51685fbf574e85))
- Fixes Nextra caching issue
- Retains caching of dependencies

Build times in our case
| Option | Build time |
| --- | --- |
| Caching enable (with problems) | 30 seconds |
| Remove `.next/cache` postbuild | 1 minute |
| VERCEL_FORCE_NO_BUILD_CACHE=1 | 3 minutes |

**Reply by juan-carlos-correa** [reply]
Adding the postbuild script solved my issue, thanks 🙏

**Comment by piclez** [comment]
@marcklingen didn't work for me, only `VERCEL_FORCE_NO_BUILD_CACHE=1`.

**Comment by xixixao** [comment]
I also ran into this, kinda baffling, but I haven't tried upgrading to latest nextra.

**Comment by atcupps** [comment]
Don't really know anything about Nextra but I had this same deployment issue with my SvelteKit project. Setting `VERCEL_FORCE_NO_BUILD_CACHE=1` worked for me as well

**Comment by juan-carlos-correa** [comment]
Same issue here, is not updating after deploy, even cleaning the browser cache. 

Wondering if this is a common issue with nextra 🤔

**Comment by athivvat** [comment]
I've found the solution to the cache issue you were experiencing with Nextra on Vercel. The Vercel build log indicates that the repository is using a shallow clone. To resolve this, please add `VERCEL_DEEP_CLONE=true` to environment variable.

**Reply by aar0npal** [reply]
Is there a build time difference between `VERCEL_DEEP_CLONE=true` and `VERCEL_FORCE_NO_BUILD_CACHE=1`?

**Reply by ChiaYuSu** [reply]
`VERCEL_FORCE_NO_BUILD_CACHE=1` is work, but `VERCEL_DEEP_CLONE=true` is not work for me.
